src/server/transaction.cc

```diff
@@ -1326,18 +1336,13 @@ void Transaction::LogAutoJournalOnShard(EngineShard* shard) {
   if (multi_ && multi_->role == SQUASHER)
     return;
 
-  bool journal_by_cmd_mask = true;
-  if ((cid_->opt_mask() & CO::NO_KEY_JOURNAL) > 0) {
-    journal_by_cmd_mask = true;  // Enforce journaling for commands that dont change the db.
-  } else if ((cid_->opt_mask() & CO::WRITE) == 0) {
-    journal_by_cmd_mask = false;  // Non-write command are not journaled.
-  } else if ((cid_->opt_mask() & CO::NO_AUTOJOURNAL) > 0 &&
-             !renabled_auto_journal_.load(memory_order_relaxed)) {
-    journal_by_cmd_mask = false;  // Command disabled auto journal.
-  }
-  if (!journal_by_cmd_mask) {
+  // Logged commands are either write commands or no-key-transactional
+  if ((cid_->opt_mask() & CO::WRITE) == 0 && (cid_->opt_mask() & CO::NO_KEY_TRANSACTIONAL) == 0)
```

Comment:
Shouldn't the second check be `> 0` ?

Reply:
No: logged commands are either WRITE or NO_KEY_TRANSACTIONAL

So if both == 0, then its neither of them, and we skip

Reply:
Ah, I understand now, its not an exclusive either, so a better phrasing would be "we log write and no-key-transactional commands" (but also assuming only one is enough 🤯 )